plugins/woocommerce/src/Admin/API/ShippingPartnerSuggestions.php

```diff
@@ -0,0 +1,212 @@
+
+		if ( $should_display ) {
+			return Suggestions::get_suggestions();
+		} elseif ( false === $should_display && true === $force_default ) {
+			return rest_ensure_response( Suggestions::get_suggestions( DefaultShippingPartners::get_all() ) );
+		}
+
+		return rest_ensure_response( Suggestions::get_suggestions( DefaultShippingPartners::get_all() ) );
+	}
+
+	/**
+	 * Get the schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public static function get_suggestions_schema() {
+		$feature_def = array(
+			'type'  => 'array',
+			'items' => array(
+				'type'       => 'object',
+				'properties' => array(
+					'icon'        => array(
+						'type' => 'string',
+					),
+					'title'       => array(
+						'type' => 'string',
+					),
+					'description' => array(
+						'type' => 'string',
+					),
+				),
+			),
+		);
+		$layout_def  = array(
+			'type'       => 'object',
+			'properties' => array(
+				'image'    => array(
+					'type'        => 'string',
+					'description' => '',
+				),
+				'features' => $feature_def,
+			),
+		);
+
+		$item_schema = array(
+			'type'       => 'object',
+			'required'   => array( 'name', 'is_visible', 'available_layouts' ),
+			// require layout_row or layout_column. One of them must exist.
+			'anyOf'      => array(
+				array(
+					'required' => 'layout_row',
+				),
+				array(
+					'required' => 'layout_column',
+				),
+			),
+			'properties' => array(
+				'name'              => array(
+					'description' => __( 'Plugin name.', 'woocommerce' ),
+					'type'        => 'string',
+					'required'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'slug'              => array(
+					'description' => __( 'Plugin slug used in https://wordpress.org/plugins/{slug}.', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'layout_row'        => $layout_def,
+				'layout_column'     => $layout_def,
+				'description'       => array(
+					'description' => __( 'Description', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'learn_more_link'   => array(
+					'description' => __( 'Learn more link .', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'is_visible'        => array(
+					'description' => __( 'Suggestion visibility.', 'woocommerce' ),
+					'type'        => 'boolean',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'available_layouts' => array(
+					'description' => __( 'Available layouts -- single, dual, or both', 'woocommerce' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type' => 'string',
+						'enum' => array( 'row', 'column' ),
```

Comment:
FYI, I borrowed `row` and `column` concept from `flex-direction`.

Reply:
I think i'd go with 'narrow' and 'wide' but no objections to this way either 😀